sources/vcdMerge/src/SignalSource.cpp

```diff
@@ -99,24 +104,27 @@ void MERGE::SignalSource::Create()
 
 void MERGE::SignalSource::SetFormat(const std::string &rFormat)
 {
-    if (rFormat == "T")
+    if (IsVcdFormat(rFormat))
     {
-        m_pSignalFactory = std::make_unique<PARSER::SignalFactory>();
+        m_pSignalFactory = std::make_unique<PARSER::DefaultSignalFactory>();
         return;
     }
 
-    if (rFormat.length() > 3)
+    if (IsUserXmlFormat(rFormat))
     {
-        if ((rFormat[0] == 'U') &&
-            (rFormat[1] == '{') &&
-            (rFormat[rFormat.length() - 1] == '}'))
-        {
-            const std::string filename(rFormat.substr(2, rFormat.length() - 3));
-            IsAccessible(filename);
+        // Extract the file name from the source description.
+        // E.g. from 'U{file_name}' the 'file_name' will be extracted.
+        // The first two characters must be skipped and the name's length does not include 'U{}'
+        const std::string filename(rFormat, 2, rFormat.length() - 3);
+        const std::ifstream infile(filename);
 
-            m_pSignalFactory = std::make_unique<PARSER::XmlSignalFactory>(filename);
-            return;
+        if (!infile.good())
+        {
+            ThrowFileInaccessibleException(filename);
```

Comment:
I don't really get the idea behind that. The part that checks if file is accessible should be a function, not `throw`.
```cpp
if (!IsAccessible(filename)) { throw; }
// ...
bool IsAccessible(const std::string &fileName)
{
    const std::ifstream inputFile(fileName);

    return inputFile.good();
}
```

Reply:
This is just this way. infile.good() checks the availability of the file. ThrowFileInaccessibleException() method just throws an exception. Nothing more. It is enclosed in the method as it is being done in two different places and I didn't want to copy the same long string in both of them.

Reply:
But file check is duplicated (**much** more problematic than duplicated string) and the logic is unclear. If the string really bothers you then formatting can be done by another function. But `if` and `throw` are part of function core logic and should be visible there.

Reply:
This can be done later anyway.